hbase/src/test/java/org/apache/metamodel/hbase/HBaseUpdateCallbackTest.java

```diff
@@ -0,0 +1,283 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.metamodel.hbase;
+
+import static org.junit.Assert.*;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.LinkedHashMap;
+import java.util.LinkedHashSet;
+import java.util.List;
+
+import org.apache.hadoop.hbase.TableName;
+import org.apache.hadoop.hbase.client.Admin;
+import org.apache.hadoop.hbase.client.Get;
+import org.apache.hadoop.hbase.client.Result;
+import org.apache.hadoop.hbase.util.Bytes;
+import org.apache.metamodel.schema.ColumnType;
+import org.apache.metamodel.schema.MutableSchema;
+import org.apache.metamodel.schema.Table;
+import org.apache.metamodel.util.SimpleTableDef;
+import org.junit.After;
+import org.junit.Before;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public abstract class HBaseUpdateCallbackTest extends HBaseTestCase {
+
+    private static final Logger logger = LoggerFactory.getLogger(HBaseClient.class);
+
+    private HBaseUpdateCallback updateCallback;
+    private MutableSchema schema;
+
+    @Override
+    @Before
+    public void setUp() throws Exception {
+        super.setUp();
+        updateCallback = new HBaseUpdateCallback(getDataContext());
+        schema = (MutableSchema) getDataContext().getDefaultSchema();
+        dropTableIfItExists();
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        dropTableIfItExists();
+    }
+
+    /**
+     * Drop the table if it exists.
+     * After that check in the schema and the datastore if the actions have been executed succesfully.
+     */
+    protected void dropTableIfItExists() {
+        if (schema != null) {
+            final Table table = schema.getTableByName(TABLE_NAME);
+            if (table != null) {
+                updateCallback.dropTable(table).execute();
+                // Check schema
+                assertNull(schema.getTableByName(TABLE_NAME));
+                // Check in the datastore
+                try (final Admin admin = getDataContext().getAdmin()) {
+                    assertFalse(admin.tableExists(TableName.valueOf(TABLE_NAME)));
+                } catch (IOException e) {
+                    fail("Should not an exception checking if the table exists");
+                }
+            }
+        }
+    }
+
+    /**
+     * Check if the table has been inserted succesfully.
+     * Checks are performed in the schema and the datastore.
+     * @throws IOException because the admin object needs to be created
+     */
+    protected void checkSuccesfullyInsertedTable() throws IOException {
+        // Check the schema
+        assertNotNull(schema.getTableByName(TABLE_NAME));
+        // Check in the datastore
+        try (final Admin admin = getDataContext().getAdmin()) {
+            assertTrue(admin.tableExists(TableName.valueOf(TABLE_NAME)));
+        } catch (IOException e) {
```

Comment:
Same here as above, don't catch the IOException here (and don't "fail" the test in case the IOException occurs).